swe_roe/ex2.c

```diff
@@ -8,418 +8,461 @@ static char help[] = "Partial 2D dam break problem.\n";
 
+/// a vector in R^3
 typedef struct {
   PetscReal V[3];
 } RDyVector;
 
+/// a type indicating one of a set of supported cell types
 typedef enum {
-  CELL_TRI_TYPE = 0, /* tetrahedron cell for a 3D cell */
-  CELL_QUAD_TYPE     /* hexahedron cell for a 3D cell */
+  CELL_TRI_TYPE = 0,  // tetrahedron cell for a 3D cell
+  CELL_QUAD_TYPE      // hexahedron cell for a 3D cell
 } RDyCellType;
 
+/// A struct of arrays storing information about mesh cells. The ith element in
+/// each array stores a property for mesh cell i.
 typedef struct {
-  PetscInt *id;         /* id of the cell in local numbering */
-  PetscInt *global_id;  /* global id of the cell in local numbering */
-  PetscInt *natural_id; /* natural id of the cell in local numbering */
-
+  /// local IDs of cells in local numbering
+  PetscInt *ids;
+  /// global IDs of cells in local numbering
+  PetscInt *global_ids;
+  /// natural IDs of cells in local numbering
+  PetscInt *natural_ids;
+
+  /// PETSC_TRUE iff corresponding cell is locally stored
   PetscBool *is_local;
 
-  PetscInt *num_vertices;  /* number of vertices of the cell    */
-  PetscInt *num_edges;     /* number of edges of the cell       */
-  PetscInt *num_neighbors; /* number of neigbors of the cell    */
-
-  PetscInt *vertex_offset;   /* vertice IDs that form the cell    */
-  PetscInt *edge_offset;     /* vertice IDs that form the cell    */
-  PetscInt *neighbor_offset; /* vertice IDs that form the cell    */
-
-  PetscInt *vertex_ids;   /* vertice IDs that form the cell    */
-  PetscInt *edge_ids;     /* edge IDs that form the cell       */
-  PetscInt *neighbor_ids; /* neighbor IDs that form the cell   */
-
-  RDyCoordinate *centroid; /* cell centroid                     */
-
-  PetscReal *area; /* area of the cell                */
-
-} RDyCell;
-
-typedef struct {
-  PetscInt *id;        /* id of the vertex in local numbering                  */
-  PetscInt *global_id; /* global id of the vertex in local numbering */
-
-  PetscBool *is_local; /* true if the vertex is shared by a local cell         */
-
-  PetscInt *num_cells; /* number of cells sharing the vertex          */
-  PetscInt *num_edges; /* number of edges sharing the vertex                   */
-
-  PetscInt *edge_offset; /* offset for edge IDs that share the vertex                       */
-  PetscInt *cell_offset; /* offset for internal cell IDs that share the vertex              */
-
-  PetscInt *edge_ids; /* edge IDs that share the vertex                       */
-  PetscInt *cell_ids; /* internal cell IDs that share the vertex              */
-
-  RDyCoordinate *coordinate; /* (x,y,z) location of the vertex                       */
-} RDyVertex;
-
-typedef struct {
-  PetscInt *id;        /* id of the edge in local numbering         */
-  PetscInt *global_id; /* global id of the edge in local numbering */
-
-  PetscBool *is_local; /* true if the edge : (1) */
-                       /* 1. Is shared by locally owned cells, or   */
```

Comment:
Do we want to keep these comments? I think those are useful.

Reply:
All these comments are intact (though maybe re-expressed) elsewhere. I've pointed you to this one.

Reply:
@jeff-cohere , thanks for the clarification.